sw/device/tests/aes/aes_test.c

```diff
@@ -52,9 +49,9 @@ int main(int argc, char **argv) {
 
   // Check against golden cipher text
   for (int i = 0; i < 16; i++) {
-    if (cipher_text_gold_32_1[i] != buffer[i]) {
-      has_error = true;
-    }
+    CHECK(cipher_text_gold_32_1[i] == buffer[i],
+          "Encoded cipher_text[%d] mismatched: exp = %x, actual = %x", i,
+          cipher_text_gold_32_1[i], buffer[i]);
```

Comment:
Seeing this in both this and your other PR makes me wonder if we want to add some variants of `CHECK` for testing equality, which would generate the CHECK message automatically... for example, for two scalars:
```
#define CHECK_EQ(_want, _got) do { \
  uint32_t want = _want, got = _got; \
  CHECK((want) == (got), "Expected equal: " #_want " = %x, " #_got " = %x.", want, got); \
} while (false)
```

Whether this is worth it probably depends on how often we intend to repeat this same style of `CHECK`...

(Also, this is not a comment that should be resolved in this PR, but mostly something to think about.)

Reply:
Yeah, we have a whole bunch of check macros in DV, that I would like to replicate for SW as well (if y'all are ok with it): 
https://github.com/lowRISC/opentitan/blob/9286ee64f9c2bbf2776d3859ee8159546d00dd96/hw/dv/sv/dv_utils/dv_macros.svh#L71

Reply:
SW is a bit more hesitant to add new macros, fwiw, but I think you should open an issue bringing it up and we can figure out if the tradeoffs of maintaining more macros is worth having more concise tests.

Reply:
Allright. 
Does the rest look ok? I hope I can merge it in today.